14/umbraco-cms/reference/custom-swagger-api.md

```diff
@@ -0,0 +1,220 @@
+
+{% endhint %}
+
+This example can be a starting point for creating a secure custom API in the Swagger documentation or using the API Controller. You can find other examples in the [API versioning and OpenAPI](./api-versioning-and-openapi.md) article.
+
+1. Create a new `.cs` file called `MyBackOfficeSecurityRequirementsOperationFilter` at the root of the project in an Umbraco v14+ installation.
+
+2. Add the following code so that the new API shows in the Swagger documentation and Swagger UI:
+
+{% code title="MyBackOfficeSecurityRequirementsOperationFilter.cs" lineNumbers="true" %}
+```csharp
+{
+using Asp.Versioning;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using Umbraco.Cms.Api.Common.Attributes;
+using Umbraco.Cms.Api.Common.Filters;
+using Umbraco.Cms.Api.Management.OpenApi;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Composing;
+using Umbraco.Cms.Core.Models.Membership;
+using Umbraco.Cms.Core.Security;
+using Umbraco.Cms.Web.Common.Authorization;
+
+namespace Umbraco.Cms.Web.UI.New.Custom;
+
+//Necessary code for the new API to show in the Swagger documentation and Swagger UI
+public class MyBackOfficeSecurityRequirementsOperationFilter : BackOfficeSecurityRequirementsOperationFilterBase
+{
+    protected override string ApiName => "my-api-v1";
+}
+
+public class MyConfigureSwaggerGenOptions : IConfigureOptions<SwaggerGenOptions>
+{
+    public void Configure(SwaggerGenOptions options)
+    {
+        options.SwaggerDoc("my-api-v1", new OpenApiInfo { Title = "My API v1", Version = "1.0" });
+        options.OperationFilter<MyBackOfficeSecurityRequirementsOperationFilter>();
+    }
+}
+
+public class MyComposer : IComposer
+{
+    public void Compose(IUmbracoBuilder builder)
+        => builder.Services.ConfigureOptions<MyConfigureSwaggerGenOptions>();
+}
+
+//Creating the Controller
+[ApiController]
+[ApiVersion("1.0")] 
+[MapToApi("my-api-v1")] 
+[Authorize(Policy = AuthorizationPolicies.BackOfficeAccess)] 
+[JsonOptionsName(Constants.JsonOptionsNames.BackOffice)]
+[Route("api/v{version:apiVersion}/my")]
+public class MyApiController : Controller
+{
+    private readonly IBackOfficeSecurityAccessor _backOfficeSecurityAccessor;
+
+    public MyApiController(IBackOfficeSecurityAccessor backOfficeSecurityAccessor)
+        => _backOfficeSecurityAccessor = backOfficeSecurityAccessor;
+
+    [HttpGet("say-hello")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
+    public IActionResult SayHello()
+    {
+        IUser currentUser = _backOfficeSecurityAccessor.BackOfficeSecurity?.CurrentUser
+                            ?? throw new InvalidOperationException("No backoffice user found");
+        return Ok($"Hello, {currentUser.Name}");
+    }
+}
+}
```

Comment:
I think this is an excess closing `}` - either that, or the entire code sample is formatted weirdly.